components/organisation-unit-tree/types/index.d.ts

```diff
@@ -0,0 +1,160 @@
+import * as React from 'react'
+
+export type OrganisationUnitTreeRoots = string | string[]
+
+interface OrganisationUnitNode {
+    id: string
+    displayName: string
+    children?: number
+    path: string
+}
+
+interface OrganisationUnitEventPayload extends OrganisationUnitNode {
+    checked: boolean
+    selected: string[]
+}
+
+export interface NodeLabelProps {
+    hasChildren: boolean
+    hasSelectedDescendants: boolean
+    loading: boolean
+    error: boolean
+    selected: string[]
+    open: boolean
+    path: string[]
+    singleSelection: boolean
+    node: OrganisationUnitNode
+    label: string
+    checked: boolean
+    highlighted: boolean
+}
+
+export interface OrganisationUnitTreeProps {
+    /**
+     * Root org unit ID(s)
+     */
+    roots: OrganisationUnitTreeRoots
+    /**
+     * Will be called with the following object:
+     * `{ id: string, displayName: string, path: string, checked: boolean, selected: string[] }`
+     */
+    onChange: (
+        payload: OrganisationUnitEventPayload,
+        event: React.MouseEvent<HTMLSpanElement>
```

Comment:
```suggestion
        event: React.MouseEvent<HTMLSpanElement | HTMLInputElement>
```

Can be triggered by clicking a checkbox as well